web/src/lib/components/resume/List.svelte

```diff
@@ -0,0 +1,107 @@
+<script lang="ts">
+    import { onDestroy, onMount } from 'svelte';
+    import { SimplePool } from 'nostr-tools';
+    import { encodeNpub } from "$lib/nostr/utils";
+    import type { UserResume } from "$lib/types/user";
+    import { subscribeResumes, subscribeMetadata, closePool } from "$lib/services/nostr";
+    import Loading from "$lib/components/Loading.svelte";
+
+    const QUERY_BATCH_SIZE = 100;
+    const CHECK_METADATA_DELAY = 1000;
+
+    export let onView = (r: UserResume, n: string | null, p: string | null) => {};
+
+    let pool = new SimplePool();
+
+    let checkMetadataTimer: ReturnType<typeof setTimeout> | null = null;
+
+    let resumes: {[pubkey: string]: {resume: UserResume, createdAt: number}} = {};
+    let metadataRequested: string[] = [];
+    let metadata: {[pubkey: string]: {name: string, picture: string}} = {};
+
+    function checkMetadataPeriodically() {
+        let pubkeysToQuery: string[] = [];
+
+        let i = 0;
+
+        for (const pubkey of Object.keys(resumes)) {
+            if (!(pubkey in metadata) && metadataRequested.indexOf(pubkey) === -1) {
+                pubkeysToQuery.push(pubkey);
+                i++;
+
+                if (i == QUERY_BATCH_SIZE) {
+                    break;
+                }
+            }
+        }
+
+        if (pubkeysToQuery.length !== 0) {
+            metadataRequested = [...metadataRequested, ...pubkeysToQuery];
+            subscribeMetadata(pool, pubkeysToQuery,
+                (pk, m) => {
+                    if (!(pk in metadata)) {
+                        metadata[pk] = {name: m.name, picture: m.picture};
+                    } else if (m.name !== undefined) {
+                        metadata[pk].name = m.name;
+                    } else if (m.picture !== undefined) {
+                        metadata[pk].picture = m.picture;
+                    }
```

Comment:
Maybe we could also get the `about` information, which could be useful to show under the name...

https://github.com/nostr-protocol/nips/blob/master/01.md#basic-event-kinds

Reply:
Definitely will do.